### ISSUE TITLE
The provided path is not a Mathcad Prime file

### ISSUE THREAD

**Issue by luretor** [MattWoodhead/MathcadPy#5]
Hey, thank you for sharing you're work!

I get this error when testing MathcadPy with your example worksheet. I'm using Mathcad Express Prime 9.0.0.
<img width="653" alt="Skjermbilde 2023-04-04 kl  22 08 44" src="https://user-images.githubusercontent.com/118355757/229911344-45906ae7-8746-4916-8582-441defd9cdf2.png">

**Comment by MattWoodhead** [comment]
Hi there!

The exception that is raised in line 91 of _application.py is as a result of failing the following if statement:

`if filepath.exists() and (filepath.suffix.lower() == ".mcdx"):`

(In the next update these are getting separated into two seperate exceptions for the file not existing, and the file extension being incorrect)

Are you certain that the file exists in the location you are passing two the mathcad_app.open() function?
It may be worth printing the full filepath in a seperate line before you pass it to the function, and then browse to the location in file explorer to check. Note that the current working directory is normally the script file location, but it is also possible to change the terminal working directory in spyder, so it is worth checking where the script is looking for the Mathcad file.

**Comment by luretor** [comment]
Hi Matt!

I've managed to get it working! It turns out the issue was indeed related to the file path.

Thanks a lot for your help! Your response was much appreciated. I should also mention that initially, I thought MathcadPy would be able to communicate with the API as long as the Mathcad file was open. 

In the future I hope I will get the chance to experiment more with integrating this into my workflow, since Mathcad has some inconvenient limitations by itself. 

 Keep up the great work!

Best regards,
luretor

**Comment by MattWoodhead** [comment]
Hi Luretor.

I am glad you got it working. Appologies for the confusion, I am trying to get sphinx up and running so I can autogenerate the documentation for the module to try and make it easier for people to get started. It is slow progress when I just work on it in my spare time I am afraid!

It is possible to access the active sheet of an already open Mathcad instance as follows, in which case you would not need to provide a filepath:

```
from MathcadPy import Mathcad

mathcad_app = Mathcad()  # If Mathcad prime is already open, this will attach to the already open instance.

ws = mathcad_app.active_sheet()  # get the active worksheet from the open instance of Mathcad

print(ws)  # The string representation of a Mathcad worksheet is the name of the file.

# more code...
```

I hope this helps.

Matt

**Comment by luretor** [comment]
Hi Matt,

Thanks for the additional information, I will check this out.

Because of my programming limitations and deadlines, I had to do my work semi-automatically within Mathcad. My calculations requires multiple iterations of the entire worksheet. For now I created a simple reference number to select which input to calculate, and which row in excel to export to. This way I could just scroll trough the calculations.

I imagine that using MathcadPy to iteratively change the reference number would be a much more efficient approach. However, for the time being, I had to make do.

Again, I appreciate all your help

Best regards,
luretor
